src/mlpack/methods/ann/layer/multihead_attention.hpp

```diff
@@ -48,54 +47,51 @@ namespace mlpack {
  * of shape `(embedDim * tgtSeqLen, batchSize)`. The embeddings are stored
  * consequently.
  *
- * @tparam InputType Type of the input data (arma::colvec, arma::mat,
- *         arma::sp_mat or arma::cube).
- * @tparam OutputType Type of the output data (arma::colvec, arma::mat,
+ * srcSeqLen is inferred from dim 0 of the previous layer.
+ * embedDim is inferred from
```

Comment:
So if we have say a 3-dimensional input, it will be [Embedding, X, Y], should the output then reshape to [Embedding, OutputSeqLen / Y, Y] or [Embedding, X, OutputSeqLen / X]?  Or just leave it as 2-d on the output.

Reply:
Obviously outputting 2-d would be the easiest -- if we add the reshape layer someone would be free to change it.  But not sure if it would be useful to say have multiple layers of these processing a 2-d (or higher) matrix of embeddings.